designdocs/provider_contexts.md

```diff
@@ -0,0 +1,339 @@
+2. Image MUST BE encoded (`png`)[https://www.w3.org/TR/png-3/] that is `250x100` (support for future sizes and light/dark should be considered).
+3. Localization keys should follow BCP 47 format like en-US, fr-FR.
+
+The data structure must support both of these and internationalization.
+
+Proposed are the following changes:
+
+1. Update `ProviderRegistryEntry`.
+
+    Each registry entry supports:
+    - A default name and logo.
+    - Localized names and logos (up to 150 locales).
+    - PNG logo support (250x100, up to 128 KiB).
+    - BCP 47-compliant language codes (e.g., en-US, fr-FR).
+2. Updated `ProviderRegistryEntry` struct have the following properties:
+
+    ```rust
+        pub struct ProviderRegistryEntry<T::Config> {
+            pub default_name: BoundedVec<u8, T::MaxProviderNameSize>,
+            pub localized_names: BTreeMap<Vec<T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderNameSize>>,
+            pub default_logo_250_100_png_bytes: BoundedVec<u8, T::MaxProviderLogo250X100Size>,
+            pub localized_logo_250_100_png_bytes: BTreeMap<Vec<T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderLogo250X100Size>>,
+        }
+    ```
+
+3. New `ProviderToApplicationRegistryEntry` storage be initialized:
+
+    ```rust
+    // Alias for clarity
+    type ApplicationIdentifier<T: Config> = BoundedVec<u8, T::MaxProviderNameSize>
+
+    #[pallet::storage]
+    pub type ProviderToApplicationRegistryEntry<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        ProviderId,
+        Twox64Concat,
+        ApplicationIdentifier<T>,
+        ProviderRegistryEntry<T>,
+        OptionQuery,
+    >;
+    ```
+
+4. `MaxProviderNameSize` be increased to `256`.
+5. `MaxProviderLogo250X100Size` be created and the limit set to `131_072` (128 KiB).
+6. Introduce hash-based logo approval mechanism:
+    Logos are not embedded during proposal submission. Instead, a blake2_256 hash of the logo image is included in the proposal.
+    Governance must explicitly approve these logo hashes as part of the provider registration process.
+
+    The approved hashes are recorded in a dedicated storage map:
+
+    ```rust
+            pub ApprovedLogoHashes: StorageMap<
+                _,
+                Blake2_128Concat,
+                [u8; 32], // blake2_256 hash of the logo
+                (),
+                OptionQuery
+            >;
+    ```
+
+    The `propose_to_be_provider` extrinsic will insert or update entries in this map.
+
+### **Changes and additions in extrinsics** <a id='extrinsics'></a>
+
+1. The `propose_to_be_provider` extrinsic will now accept an optional list of hashes for images/logos to be approved by governance.
+
+    ```rust
+        #[pallet::call_index(0)]
+        pub fn propose_to_be_provider(
+            origin: OriginFor<T>,
+            provider_name: Vec<u8>,
+            logo_hashes: Vec<[u8; 32]>, // blake2_256 hashes of the logos
```

Comment:
Question: do we also take a list of localized names to be approved by goverance? or create a typed payload like `ProviderRegisterEntry` but with hashes instead of image bytes?